### PR TITLE
Support ListObjectsWithMetadata API

### PR DESCRIPTION
**Pull request by harshavardhana**

### CODE REVIEW

**Comment by harshavardhana** [comment]
Of course, needs server PR https://github.com/minio/minio/pull/8539

File: api-list.go [hunk 1]
```diff
@@ -319,7 +360,7 @@ func (c Client) ListObjects(bucketName, objectPrefix string, recursive bool, don
 		var marker string
 		for {
 			// Get list of objects a maximum of 1000 per request.
```
Comment (kannappanr) [inline api-list.go]: What is the maximum returned list size per request? Is it 10000?

File: api-list.go [hunk 2]
```diff
@@ -600,11 +641,10 @@ func (c Client) listMultipartUploadsQuery(bucketName, keyMarker, uploadIDMarker,
 	urlValues.Set("encoding-type", "url")
 
 	// maxUploads should be 1000 or less.
-	if maxUploads == 0 || maxUploads > 1000 {
-		maxUploads = 1000
+	if maxUploads > 0 {
```
Comment (kannappanr) [inline api-list.go]: Should there be a max number?

File: api-list.go (hunk 1 above)
Comment (harshavardhana) [reply api-list.go]: maximum depends on server implementation @kannappanr

File: api-list.go (hunk 2 above)
Comment (harshavardhana) [reply api-list.go]: maximum depends on server implementation @kannappanr

**Pull request merged**